WooCommerce/Classes/ViewRelated/Orders/Order Filters/Date Range Filter/OrderDatesFilterViewController.swift

```diff
@@ -0,0 +1,182 @@
+        tableView.deselectRow(at: indexPath, animated: true)
+
+        switch rows[indexPath.row].filter {
+        case .custom:
+            // Open the View Controller for selecting a custom range of dates
+            //
+            let dateRangeFilterVC = DateRangeFilterViewController(startDate: selected?.startDate,
+                                                                  endDate: selected?.endDate) { [weak self] (startDate, endDate) in
+                guard let self = self else { return }
+                self.selected = OrderDateRangeFilter(filter: .custom, startDate: startDate, endDate: endDate)
+                self.configureRows()
+                self.onCompletion(self.selected)
+                self.tableView.reloadData()
+            }
+            navigationController?.pushViewController(dateRangeFilterVC, animated: true)
+        case .any:
+            selected = nil
+            onCompletion(nil)
+            configureRows()
+            tableView.reloadData()
+        default:
+            selected = rows[indexPath.row]
+            onCompletion(selected)
+            configureRows()
+            tableView.reloadData()
+            return
+        }
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+}
+
+// MARK: - Cell configuration
+//
+private extension OrderDatesFilterViewController {
+    /// Cells currently configured in the order they appear on screen
+    ///
+    func configure(_ cell: UITableViewCell, for row: OrderDateRangeFilter, at indexPath: IndexPath) {
+        switch cell {
+        case let cell as BasicTableViewCell:
+            configureStandardFilter(cell: cell, row: row)
+        case let cell as TitleAndValueTableViewCell where row.filter == .custom:
+            configureCustomDateRangeFilter(cell: cell, row: row)
+        default:
+            fatalError()
+            break
+        }
+    }
+
+    func configureStandardFilter(cell: BasicTableViewCell, row: OrderDateRangeFilter) {
+        cell.textLabel?.text = row.description
+        cell.selectionStyle = .none
+        if row.filter == selected?.filter || row.filter == .any && selected == nil {
+            cell.accessoryType = .checkmark
+        }
+        else {
+            cell.accessoryType = .none
+        }
+    }
+
+    func configureCustomDateRangeFilter(cell: TitleAndValueTableViewCell, row: OrderDateRangeFilter) {
+        var formattedDates: String = ""
+        if let startDateFormatted = row.startDate?.toString(dateStyle: .medium, timeStyle: .none) {
+            formattedDates = startDateFormatted
+        }
+        if row.startDate != nil && row.endDate != nil {
+            formattedDates = formattedDates + " - "
+        }
+        if let endDateFormatted = row.endDate?.toString(dateStyle: .medium, timeStyle: .none) {
+            formattedDates = formattedDates + endDateFormatted
+        }
+
```

Comment:
How about using [`DateIntervalFormatter` for the date range string](https://developer.apple.com/documentation/foundation/dateintervalformatter), so that it can be localized automatically if both dates are non-nil?

Reply:
I didn't used the `DateIntervalFormatter` because here we can select just the start date or just the end date, and the DateIntervalFormatter accept only non optionals value, maybe you had in mind another way to use `DateIntervalFormatter`?

Reply:
I was thinking to use `DateIntervalFormatter` when both dates are non-nil, otherwise just the date string 😄